dlrover/python/unified/common/workload_base.py

```diff
@@ -51,7 +53,7 @@ class JobInfo:
 
     name: str
     job_id: str
-    user_config: dict
+    user_config: Union[dict, DictConfig]
```

Comment:
It's only Dict[str,Any], can't be DictConfig. As pydantic don't support DictConfig.